projects/engine/src/include/Core/GameWorld.h

```diff
@@ -86,23 +85,22 @@ namespace MAD
 		if (targetWorldLayerIter == m_worldLayers.end())
 		{
 			// Create new world layer and assign its owning world
-			targetWorldLayerIter = m_worldLayers.emplace(inWorldLayerName).first;
+			targetWorldLayerIter = m_worldLayers.emplace(inWorldLayerName, OGameWorldLayer(this)).first;
 			targetWorldLayerIter->second.SetWorldLayerName(inWorldLayerName);
-			targetWorldLayerIter->second.SetOwningWorld(*this);
 		}
 
 		LOG(LogDefault, Log, "Spawning Entity at Layer: %s\n", targetWorldLayerIter->first.c_str());
 
 		// Create default EntityType object through common creation API and assign the entity's owning world layer
-		eastl::shared_ptr<CommonAncestorEntityType> defaultEntityObject = inTypeInfo.CreateDefaultObject<CommonAncestorEntityType>();
+		eastl::shared_ptr<CommonAncestorEntityType> defaultEntityObject = inTypeInfo.CreateDefaultObject<CommonAncestorEntityType>(this);
 		
-		//Add entity to the layer's entity list
 		targetWorldLayerIter->second.AddEntityToLayer(defaultEntityObject);
 
-		// Since we defer the registration of the entity's owning GameWorldLayer and it's owning GameWorld, you cannot access these within the constructor of the entity
 		RegisterEntity(*defaultEntityObject, targetWorldLayerIter->second);
 
-		defaultEntityObject->OnBeginPlay();
+		defaultEntityObject->OnBeginPlay(); // TODO: When should this happen (?)
+
+		defaultEntityObject->PostInitializeComponents(); // TODO: When should this happen (?)
```

Comment:
This should happen here (at the end of this function that is), but before `OnBeginPlay`